nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -783,6 +783,7 @@ def configure_optimizers(self):
                 fp32_grad_accum = False
                 # TODO: contiguous grad bucket for fp16 is also planned to be supported
                 contiguous_grad_bucket = False
+                raise ValueError("fp16 trained is not yet supported with O2. Please use O1.")
```

Comment:
I think you mean "training" and not "trained"? Also from a user point of view, we should probably is not supported with `megatron_amp_O2=True`, please set it to False or something like that?

Reply:
Done.